api/dp-openapi.yaml

```diff
@@ -917,6 +917,137 @@ paths:
         default:
           $ref: '#/components/responses/UnexpectedError'
 
+  /projects/{projectid}/checkout:
+    put:
+      tags:
+      - project
+      description: Check out a page for proofreading. Response - schema - proof_page.
+      parameters:
+      - $ref: '#/components/parameters/projectid'
+      - $ref: '#/components/parameters/state_query'
+      responses:
+        200:
+          description: page data
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/proof_page'
+        '400':
+          $ref: '#/components/responses/InvalidValue'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '429':
+          $ref: '#/components/responses/RateLimitExceeded'
+        default:
+          $ref: '#/components/responses/UnexpectedError'
+
+  /projects/{projectid}/validatetext:
+    put:
+      tags:
+      - project
+      description: Checks the given text for invalid characters, returns data indicating them
+      parameters:
+      - $ref: '#/components/parameters/projectid'
+      requestBody:
+        $ref: '#/components/requestBodies/page_text'
+      responses:
+        200:
+          description: analysis data
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/text_analysis'
+        '400':
+          $ref: '#/components/responses/InvalidValue'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '429':
+          $ref: '#/components/responses/RateLimitExceeded'
+        default:
+          $ref: '#/components/responses/UnexpectedError'
+
+  /projects/{projectid}/pages/{pagename}:
+    put:
+      tags:
+      - project
+      description: Various operations on the page as determined by the 'pageaction_query' parameter
+      parameters:
+      - $ref: '#/components/parameters/projectid'
+      - $ref: '#/components/parameters/state_query'
+      - $ref: '#/components/parameters/pagename'
+      - $ref: '#/components/parameters/pagestate_query'
+      - name: pageaction_query
+        in: query
+        description: Action to perform on the page.
+            <br><br>
+            resume - If the page is saved-as-done 'unsaves' it. Response - schema - proof_page.
```

Comment:
```suggestion
            resume - If the page is saved-as-done 'unsaves' it. Response: proof_page object.
```

Reply:
Unfortunately the colon upsets the swagger editor.